docs/js/immutablejs.md

```diff
@@ -11,66 +11,78 @@ for a good explanation of the more intricate benefits it has.
+For instance you'll write `map.get('property')` instead of `object.property`, and `list.get(0)` instead of `array[0]`.  
+This can make your code a little harder to follow and requires you to be extra cautious when passing arguments or props to functions or components that try to access values with regular dot notation.  
+ImmutableJS's [`Record`](https://facebook.github.io/immutable-js/docs/#/Record) structure offers a solution to this issue.
 
-That means you must use getters to access properties : for instance you'll do `map.get("property")` instead of `map.property`, and `array.get(0)` instead of `array[0]`. It's not natural and your code becoming bigger, you finish by not knowing anymore if you are working with a JavaScript object or an Immutable one. While it's possible to be clear where you are using immutable objects, you still pass them through the system into places where it's not clear. This makes reasoning about functions harder.
+A `Record` is similar to a `Map` but has a fixed shape, meaning it's property keys are predefined and you can't later add a new property after the record is created.  
+One benefit of `Record` is that you can now, along with other immutable read methods (.get, .set, .merge and so on), use the dot notation to access properties.
 
-The `Record` structure tries to get rid of this drawback. `Record` is like a `Map` whose shape is fixed : you can't later add a new property after the record is created. The benefit of `Record` is that you can now, along with others .get, .set and .merge methods, use the dot notation to access properties, which is a good point to write simpler code.
-
-The creation of a record is less simple. You got to first create the `Record` shape. With the example above, to create your initial state, you'll write :
+The creation of a record is less simple than simply calling `.toJS()`.  
+First, you have to define the `Record` shape. With the example above, to create your initial state, you'll write:
 
 ```JS
 //the shape
 const StateRecord = Record({
-  myData: 'Hello World!',
+  myData: {
+      message: 'Hello World!'
+  }
 });
 
 const initialState = new StateRecord({}); // initialState is now a new StateRecord instance
-                                          // initialized with myData set by default as 'Hello World!'
+                                          // initialized with myData.message set by default as 'Hello World!'
 ```
 
-Now, if you want to access `myData`, you can just write `state.myData` in your reducer code.
+Now, if you want to access `myData`, you can just write `state.myData` in your reducer code and to access the `message` property you can write `state.myData.message` as you would in a plain javascript object.
 
+It's worth noting that although dot notation can now be used to read properties the same does not apply to setting properties. Any attempts to set a property on a `Record` using dot notation will result in errors.
```

Comment:
Yeah... Record extends `Collection` and so inherits the methods from `Iterable` but you'd never know looking at the docs (which btw, I often use the canonical example of why automated docs suck!)

Relatedly, the `size` thing is a doozy: really common key in incoming payloads, really baffling (and destructive) behaviour. I say relatedly because the error derives from the collision with the built-in `Collection.size` property!
Again nothing in the docs. I asked Lee Byron directly on Twitter what the deal was and he said the v4 would fix it... but there doesn't seem to have been any activity on Immutable for months now, so... look out I guess?